src/cljs/replumb/repl.cljs

```diff
@@ -108,20 +109,43 @@
   [form]
   (and (seq? form) (replumb-repl-special-set (first form))))
 
+(defn make-js-eval-fn
+  "Makes an eval function that will be used to eval JavaScript code. It returns
+  a cljs.js-compatible *eval-fn*. Expects a map of user options, specifically:
+
+  * :cache-path - if present, the function will first write the cache files and
+  then eval the source, otherwise only the latter
+  * write-file-fn! - synchronous 2-arity function which expects the path and data
+  to write."
+  [{:keys [cache-path write-file-fn! verbose]}]
+  (fn [{:keys [path name source cache]}]
+    (when (and path source cache cache-path)
```

Comment:
I would check also that `write-file-fn!` is not nil or it will blow up (with no trace btw)